containerd-http-proxy/configure_http_proxy.yaml

```diff
@@ -0,0 +1,107 @@
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: containerd-http-proxy
+spec:
+  selector:
+    matchLabels:
+      name: containerd-http-proxy
+  template:
+    metadata:
+      labels:
+        name: containerd-http-proxy
+    spec:
+      hostPID: true
+      containers:
+      - name: startup-script
+        image: gke.gcr.io/debian-base:bookworm-v1.0.0-gke.1
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          privileged: true
+        command:
+        - /bin/sh
+        - -c
+        - |
+          set -e
+          set -u
+
+          validate_proxy() {
+            input_string=$1
+
+            if echo "$input_string" | grep -qvE '^(HTTP_PROXY|HTTPS_PROXY|NO_PROXY)='; then
```

Comment:
As this is implemented now, customer can set environment variable we do not intent. I didn't test it, but this doc: https://www.freedesktop.org/software/systemd/man/latest/systemd.service.html suggests that a single `Environment=` can be set to declare multiple variables. So the value of `HTTP_PROXY="http://proxy.example.com:80" ANOTHER_VAR="BAR"` will work

Reply:
Added quote check in validation.